### PR TITLE
[DS-4118] change_password and register emails have hardcoded sample addresses

### PR DESCRIPTION
**Pull request by mwoodiupui**
https://jira.duraspace.org/browse/DS-4118
Fixes #7465
Replace hard-coded email address and telephone number with DSpace configuration properties, so that all affected templates can be customized by the same simple edits to dspace.cfg.

Also fix a typo in DS-3872.

Tested using `bin/dspace test-email`.

### CODE REVIEW

**Review by tdonohue — COMMENTED** [review]
@mwoodiupui : Gave this a quick review, and it looks fine to me overall.

That said, one simple/obvious question arises.  Do many sites actually *use* a helpdesk telephone number?  A part of me wonders if we should instead consider updating these templates to just say "please contact us at ${config.get("mail.helpdesk")}."  and leave it at that.  Individual sites obviously can add a contact telephone number if they want one.

I honestly don't have a strong opinion. I just wanted to ask the question.  One more minor suggestion inline.

File: dspace/config/dspace.cfg
Comment (tdonohue) [inline dspace/config/dspace.cfg]: Should we consider renaming this configuration something like `mail.message.telephone` since it's only used in mail templates? I'm also fine with it named as-is, but we probably should make it clear in the comment that it's only used by the mail templates, and not displayed in UIs, etc.

**Comment by mwoodiupui** [comment]
A good point.  I think the template could be enhanced to omit the "or call us at..." clause if the phone number is not defined.  But it would be good to discuss what should be in these templates.  Some of them look as though they haven't been touched since the days when there was only one DSpace instance in the universe.

File: dspace/config/dspace.cfg
Comment (mwoodiupui) [reply dspace/config/dspace.cfg]: I didn't like the name either, but also disliked all the alternatives I had thought of up to then.  `mail.helpdesk.telephone`?

Comment (mwoodiupui) [reply dspace/config/dspace.cfg]: I would argue that the helpdesk phone number might be thought by some to be useful in UI pages and adopted for local use there, so we shouldn't be too quick to comment that it isn't.  But I've renamed it and made it optional.

**Review by tdonohue — APPROVED** [review]
👍 Looks good to me. I have not had a chance to test it yet though.

**Comment by mwoodiupui** [comment]
Close/open to get Travis to run.

**Comment by mwoodiupui** [comment]
Kick the test runner again.

**Review by tdonohue — APPROVED** [review]
👍 @mwoodiupui : Had a chance to re-review this today, and it looks good. I tested by registering a few new users & I can verify that the phone is only displayed when configured.

One thing I did want to point out, which might require minor code changes... I realized on re-review that this PR essentially changes the behavior of the existing `mail.helpdesk` setting.

* Before this PR , that setting is only used by "Request a Copy", so it's located in that section of dspace.cfg: https://github.com/DSpace/DSpace/blob/c36a901170e4ae8b8de5707301b06f1942333009/dspace/config/dspace.cfg#L1526-L1528
* With this PR, that `mail.helpdesk` is now used more as a general Help Desk email.  This means you've chosen to use it in email templates where a Help Desk might exist...while the `mail.admin` is now only used for more technical emails. 

This change means that `mail.helpdesk` becomes a more generic configuration.  That also likely means we may want to move `mail.helpdesk` and `mail.helpdesk.name` up in the `dspace.cfg` file to the "Email settings" section (perhaps near the `mail.admin` or similar): https://github.com/DSpace/DSpace/blob/c36a901170e4ae8b8de5707301b06f1942333009/dspace/config/dspace.cfg#L121 

That's a very tiny change (and obviously won't impact the behavior of this PR). But it seems like it might be worth that small re-org before we merge this, assuming you agree.

**Comment by tdonohue** [comment]
Thanks @mwoodiupui !  This looks good to me now.  Merging

**Pull request merged**
